src/python/zquantum/qaoa/recursive_qaoa.py

```diff
@@ -95,25 +86,47 @@ def _minimize(
                     each tuple is a tuple of bits.
         """
 
+        n_qubits = count_qubits(
+            change_operator_type(self._cost_hamiltonian, QubitOperator)
+        )
+        qubit_map = _create_default_qubit_map(n_qubits)
+
+        histories: Dict[str, List[HistoryEntry]] = defaultdict(list)
+        histories["history"] = []
+
+        return self._recurse(
+            cost_function_factory,
+            initial_params,
+            keep_history,
+            cost_hamiltonian=self._cost_hamiltonian,
+            qubit_map=qubit_map,
+            nit=0,
+            nfev=0,
+            histories=histories,
+        )
+
+    def _recurse(
```

Comment:
I would rename it to something more descriptive. Perhaps `_recursive_minimization` or sth like that?